kraft-jennifer-ndvi-automation.py

```diff
@@ -0,0 +1,698 @@
+# 
+# 
+# ## Assignment Requirements
+# 
+# Your submission to the GitHub repository should include:
+# * This Jupyter Notebook file (.ipynb) with:
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
+# # Pseudocode workflow
+#  
+# The workflow for this notebook consists of the following steps:
+# 
+#  Importing the necessary packages.
+#  Defining the functions needed for 1) for opening and cleaning, and 2) cloud masking. 
+#  
+# The function for opening and cleaning is kept as clean and consize as possible, and is optimized to run quickly/efficiently by using a loop for creating a list of directories, and generating site names from the file path.
+# 
+# The .shp files and crop boundaries are opened; each needed band (.tif) is opened, clipped, then cleaned with pixel qa.
+# 
+# NDVI is calculated, NDVI layer clipped, and cloud mask applied to NDVI layer.
+# 
+# After cloud removal, NDVI Averages are calculated, then plotted for each site, with data averages as collected between January through December of 2017.
+# 
+# 
+
+# In[1]:
+
+
+# Autograding imports - do not modify this cell
+import matplotcheck.autograde as ag
+import matplotcheck.notebook as nb
+import matplotcheck.timeseries as ts
+from datetime import datetime
+
+
+# In[2]:
+
+
+# Import needed packages in PEP 8 order
+# and no unused imports listed (10 points total)
+
+# YOUR CODE HERE
+
+# Import necessary packages
+import os
+from glob import glob
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+
+import rioxarray as rxr
+import xarray as xr
+from rasterio.plot import plotting_extent
+
+import earthpy as et
+import earthpy.spatial as es
+import earthpy.plot as ep
+
+
+os.chdir(os.path.join(et.io.HOME,
```

Comment:
Yes - paths created to ensure they work on all operating systems (using os.path.join) (Reproducibility)